tests/spec/fs.rename.spec.js

```diff
@@ -70,6 +70,33 @@ describe('fs.rename', function() {
     });
   });
 
+  it('should rename an existing directory (using promises)', () => {
+    var fsPromises = util.fs().promises;
+
+    return fsPromises.mkdir('/mydir')
+      .then(() => {
+        return fsPromises.rename('/mydir', '/myotherdir')
+          .catch((error) => expect(error).not.to.exist);
+      })
+      .then(() => {
+        return fsPromises.stat('/mydir')
+          .catch((error) => {
+            expect(error).to.exist;
+            expect(error.code).to.equal('ENOENT');
+          });
+      })
+      .then(() => {
+        return fsPromises.stat('/myotherdir')
+          .catch((error, result) => {
+            expect(error).not.to.exist;
```

Comment:
Same as above: you would expect an error to exist if you are catching it
You probably want something along these lines:

```js
.then(() => fsPromises.stat('/myotherdir'))
.then((error, result) => {
     expect(error).not.to.exist;
     expect(result.nlinks).to.equal(1);
})

Reply:
Instead of doing `.catch((error, result) => {` you can do:

```js
.then(result => {
  expect(result.nlinks).to.equal(1);
}
```